src/status_im/contexts/profile/events.cljs

```diff
@@ -39,32 +40,45 @@
 (rf/reg-event-fx
  :profile/profile-selected
  (fn [{:keys [db]} [key-uid]]
-   {:db (update db :profile/login #(select-profile % key-uid))}))
+   {:db (update db :profile/login select-profile key-uid)}))
+
+(rf/reg-event-fx
+ :profile/set-profile-overview-auth-method
+ (fn [{db :db} [key-uid auth-method]]
+   {:db (assoc-in db [:profile/profiles-overview key-uid :auth-method] auth-method)}))
+
+(rf/reg-event-fx
+ :profile/get-profiles-auth-method
+ (fn [_ [key-uids]]
+   (let [auth-method-fx (fn [key-uid]
+                          [:effects.biometric/check-if-available
+                           {:key-uid    key-uid
+                            :on-success (fn [auth-method]
+                                          (rf/dispatch
+                                           [:profile/set-profile-overview-auth-method
+                                            key-uid
+                                            auth-method]))}])]
+     {:fx (map auth-method-fx key-uids)})))
 
 (rf/reg-event-fx
  :profile/get-profiles-overview-success
- (fn [{:keys [db]} [{:keys [accounts] {:keys [userConfirmed enabled]} :centralizedMetricsInfo}]]
-   (let [db-with-settings  (assoc db
-                                  :centralized-metrics/user-confirmed? userConfirmed
-                                  :centralized-metrics/enabled?        enabled)
-         profiles          (reduce-profiles accounts)
-         {:keys [key-uid]} (first (sort-by :timestamp > (vals profiles)))
-         new-db            (cond-> db-with-settings
-                             (seq profiles)
-                             (assoc :profile/profiles-overview profiles)
+ (fn [{:keys [db]}
+      [{accounts                        :accounts
+        {:keys [userConfirmed enabled]} :centralizedMetricsInfo}]]
+   (let [profiles          (reduce-profiles accounts)
```

Comment:
I've read a few times the recommendation and tried to follow myself when appropriate that it's _sometimes better_ to abstract computations per unit, not collection. This approach tends to produce code that's more reusable.

In this instance, one alternative would be:

```clojure
(defn index-as-profile-overview
  [acc {:keys [key-uid] :as profile}]
  (assoc acc key-uid (profile.rpc/rpc->profiles-overview profile)))

(rf/reg-event-fx
 :profile/get-profiles-overview-success
 (fn [{:keys [db]}
      [{accounts                        :accounts
        {:keys [userConfirmed enabled]} :centralizedMetricsInfo}]]
   (let [profiles          (reduce index-as-profile-overview {} accounts)
         ...]
     {...})))
```

Reply:
Yeah, I agree.

I kept the `reduce-profiles` function exactly as before, I didn't reworked it because it's not too related, ofc, I can apply your suggestion :+1: